# LLM Evaluation Framework — Reviewed Results Walkthrough

This companion notebook reads the project's committed evaluation summary and explains the result immediately after each output. The evaluation contains 20 medical-QA questions across five domains, so conclusions should be treated as a focused experiment rather than a population-level benchmark.


## 1. Reproduce the committed overall summary


In [1]:
# In the repository this is stored in eval_summary.json.
overall = {
    "base_avg_rouge": 0.1971,
    "ft_avg_rouge": 0.2321,
    "improvement_pct": 17.8,
    "win_rate_pct": 65.0,
    "total_questions": 20,
}
for key, value in overall.items():
    print(f"{key}: {value}")


base_avg_rouge: 0.1971
ft_avg_rouge: 0.2321
improvement_pct: 17.8
win_rate_pct: 65.0
total_questions: 20


### What this tells us

The fine-tuned model has higher average ROUGE and wins 13 of 20 questions. The **17.8% improvement is relative to the base model's average ROUGE**, while the 65% win rate shows the advantage is not universal—roughly one third of questions do not favor the tuned model.


## 2. Compare performance by clinical category


In [1]:
# Category-level results expose where the aggregate improvement comes from.
base = {
    "Anatomy & Physiology": 0.1749,
    "Pathophysiology": 0.1228,
    "Pharmacology": 0.2280,
    "Symptoms & Diagnosis": 0.1868,
    "Treatment": 0.2728,
}
finetuned = {
    "Anatomy & Physiology": 0.2272,
    "Pathophysiology": 0.1816,
    "Pharmacology": 0.2764,
    "Symptoms & Diagnosis": 0.1928,
    "Treatment": 0.2823,
}
for category in base:
    delta = finetuned[category] - base[category]
    print(f"{category:22s} base={base[category]:.4f}  tuned={finetuned[category]:.4f}  delta={delta:+.4f}")


Anatomy & Physiology   base=0.1749  tuned=0.2272  delta=+0.0523
Pathophysiology        base=0.1228  tuned=0.1816  delta=+0.0588
Pharmacology           base=0.2280  tuned=0.2764  delta=+0.0484
Symptoms & Diagnosis   base=0.1868  tuned=0.1928  delta=+0.0060
Treatment              base=0.2728  tuned=0.2823  delta=+0.0095


### What this tells us

Every category improves in the committed summary, but the gains are highly uneven. Pathophysiology, anatomy/physiology, and pharmacology drive most of the improvement; symptoms/diagnosis and treatment change only slightly. Looking only at the overall average would hide that pattern.


## 3. Keep the metric limitation visible


In [1]:
# A win-rate complement makes the regression rate explicit.
win_rate = 65.0
print(f"fine_tuned_win_rate: {win_rate:.1f}%")
print(f"non_win_rate: {100 - win_rate:.1f}%")


fine_tuned_win_rate: 65.0%
non_win_rate: 35.0%


### What this tells us

A 35% non-win rate is a reminder that fine-tuning does not improve every prompt. ROUGE also measures lexical overlap, not clinical correctness, factuality, safety, or reasoning quality. The framework is useful evidence of directional improvement, but it should be extended with expert or rubric-based evaluation before stronger medical claims.


## Final interpretation

This project is strongest as an **evaluation framework**: it preserves per-question results, category summaries, a base-vs-tuned comparison, and an interactive dashboard. The next maturity step would be to add semantic/factual scoring and human review alongside ROUGE.
